src/main/java/seedu/sudohr/storage/JsonSerializableSudoHr.java

```diff
@@ -47,13 +54,23 @@ public JsonSerializableSudoHr(ReadOnlySudoHr source) {
      */
     public SudoHr toModelType() throws IllegalValueException {
         SudoHr sudoHr = new SudoHr();
+
         for (JsonAdaptedPerson jsonAdaptedPerson : persons) {
             Person person = jsonAdaptedPerson.toModelType();
             if (sudoHr.hasPerson(person)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_PERSON);
             }
             sudoHr.addPerson(person);
         }
+
+        for (JsonAdaptedDepartment jsonAdaptedDepartment : departments) {
```

Comment:
Not handled: Possibly duplicate person (same ID) or person with same phone/email as another in a department but this will be handled on my end when i re-define equality between two persons. Data added to SudoHR above enforce these checks.